t/object_methods.t

```diff
@@ -2,86 +2,113 @@ use strict;
 
-# plus_time_interval, minus_time_interval
-is($datetime2->plus_time_interval('1d')->is_same_as($datetime3),   1,          'plus_time_interval("1d") yields one day ahead.');
-is($datetime1->plus_time_interval(0),                              $datetime1, 'plus_time_interval(0) yields the same object');
-is($datetime3->plus_time_interval('-1d')->is_same_as($datetime2),  1,          'plus_time_interval("-1d") yields one day back.');
-is($datetime3->minus_time_interval('1d')->is_same_as($datetime2),  1,          'minus_time_interval("1d") yields one day back.');
-is($datetime1->minus_time_interval(0),                             $datetime1, 'minus_time_interval(0) yields the same object');
-is($datetime2->minus_time_interval('-1d')->is_same_as($datetime3), 1,          'minus_time_interval("-1d") yields one day ahead.');
-throws_ok { $datetime3->minus_time_interval("one") } qr/Bad format/, 'minus_time_interval("one") is not a mind-reader..';
+subtest 'truncate_to_day' => sub {
+    is($datetime1->truncate_to_day->datetime_iso8601, "2011-12-13T00:00:00Z", "Truncates time correctly");
+    is($datetime1->truncate_to_day->is_same_as($datetime2->truncate_to_day), 1,     "is_same_as for truncated objects on the same day");
+    is($datetime2->truncate_to_day->is_same_as($datetime3->truncate_to_day), undef, "is_same_as for truncated objects on the different days");
+};
+
+subtest 'plus_time_interval' => sub {
+    is($datetime2->plus_time_interval('1d')->is_same_as($datetime3),  1,          'plus_time_interval("1d") yields one day ahead.');
+    is($datetime1->plus_time_interval(0),                             $datetime1, 'plus_time_interval(0) yields the same object');
+    is($datetime3->plus_time_interval('-1d')->is_same_as($datetime2), 1,          'plus_time_interval("-1d") yields one day back.');
+};
+
+subtest 'minus_time_interval' => sub {
+    is($datetime3->minus_time_interval('1d')->is_same_as($datetime2),  1,          'minus_time_interval("1d") yields one day back.');
+    is($datetime1->minus_time_interval(0),                             $datetime1, 'minus_time_interval(0) yields the same object');
+    is($datetime2->minus_time_interval('-1d')->is_same_as($datetime3), 1,          'minus_time_interval("-1d") yields one day ahead.');
+    throws_ok { $datetime3->minus_time_interval("one") } qr/Bad format/, 'minus_time_interval("one") is not a mind-reader..';
+};
+
+subtest 'move_to_nth_dow' => sub {
+    is($datetime1->move_to_nth_dow(3,   'Wed')->day_of_month,      21,    'Third Wednesday of Dec 2011 is the 21st');
+    is($datetime1->move_to_nth_dow(5,   'SuNdaY'),                 undef, '... and there is no 5th Sunday that year.');
+    is($datetime1->move_to_nth_dow(-50, 'Mon'),                    undef, '... nor a -50th Monday.');
+    is($datetime1->move_to_nth_dow(105, 'Tue'),                    undef, '... nor a 105th Tuesday.');
```

Comment:
What I tried to say is: It's perhaps also a good idea to test for cases that only just exceed the boundaries. `-50` and `105` are way off. But `0` is just out of bounds. Also, there is a 5th Wed in Sep 2015 but not in Aug 2015.

This kind of testing is not so much to make sure the current implementation works as expected but also against future refactoring.